src/lib/icon/icon.ts

```diff
@@ -236,7 +240,19 @@ export class MatIcon extends _MatIconMixinBase implements OnChanges, OnInit, Can
       this._previousFontIconClass = this.fontIcon;
     }
   }
-
+ 
+  /**
+   * Updates the url paths with current paths append to icon which fixes
+   * SVG filters in Safari/Firefox
+   */
+  private _updateUrlPaths(svg: SVGElement) {
+    const currentPath = this._location.prepareExternalUrl(this._location.path());
+    
+    svg.outerHTML = svg.outerHTML.replace(/url\((.*)\)/, `url(${currentPath}$1)`);
```

Comment:
This seems very inefficient since it's asking the browser to parse the HTML once and then re-parse it once the URLs are updated. Since you're using a regex anyway, why not do it before the element is even created?

Reply:
Regex does seem a bit brittle I agree. I'll admit I didn't read through the code at all, so I probably could of found a better spot. When I get a few minutes somehow I'll read the code and find a better spot for this!